File: lib/build_targets/package.dart
// Copyright 2026 The FlutterWatch Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'package:flutter_tools/src/build_system/build_system.dart';

class NativeWatchosPluginPackage extends Target {
  NativeWatchosPluginPackage();

  @override
  String get name => 'watchos_native_plugin_package';

  @override
  List<Target> get dependencies => const <Target>[];

  @override
  List<Source> get inputs => const <Source>[];

  @override
  List<Source> get outputs => const <Source>[];

  @override
  Future<void> build(Environment environment) async {
    // Build package target logic if applicable
  }
}

File: packages/flutter_watchos/watchos/Package.swift
// swift-tools-version:5.9
// Copyright 2026 The FlutterWatch Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import PackageDescription

let package = Package(
    name: "flutter_watchos",
    platforms: [
        .watchOS(.v7),
    ],
    products: [
        .library(name: "flutter-watchos", targets: ["flutter_watchos"]),
    ],
    targets: [
        .target(
            name: "flutter_watchos",
            path: "Classes",
            publicHeadersPath: ".",
            cSettings: [
                .headerSearchPath("."),
            ],
            linkerSettings: [
                .linkedFramework("WatchKit"),
                .linkedFramework("Foundation"),
            ]
        ),
    ]
)
